magma_ff/create_metawflrun_handler.py

```diff
@@ -0,0 +1,358 @@
+            meta_workflow_run_step[ITEMS_FOR_CREATION] = self._embed_items_for_creation(
+                meta_workflow_step
+            )
+
+            # Basic dict for current MetaWorkflow Run step complete. Now append.
+            ordered_meta_workflow_runs.append(meta_workflow_run_step)
+
+        return ordered_meta_workflow_runs
+
+    def _embed_items_for_creation(self, meta_workflow_step):
+        """
+        From a MetaWorkflow Step, extract the items_for_creation attribute, which
+        may be uuids or property traces (in relation to the associated item).
+
+        If uuids, return this list of uuids.
+        If property traces, use embed requests to convert to identifiers.
+
+        :param meta_workflow_step: object containing a MetaWorkflow Step's metadata
+        :type meta_workflow_step: dict
+        :return: list of items_for_creation identifiers
+        :rtype: list[str]
+        :raises MetaWorkflowRunHandlerCreationError: if a property trace cannot be embedded
+        """
+        # if items_for_creation_uuid, just copy over
+        # if ITEMS_FOR_CREATION_UUID in meta_workflow_step.keys():
+        if getattr(meta_workflow_step, ITEMS_FOR_CREATION_UUID, None):
+            return getattr(meta_workflow_step, ITEMS_FOR_CREATION_UUID)
+        # otherwise, dealing with property traces. Make necessary embed requests
+        # and convert property trace(s) to uuid(s)
+        else:
+            property_traces = getattr(meta_workflow_step, ITEMS_FOR_CREATION_PROP_TRACE, None)
+            if not isinstance(property_traces, list):
+                item_uuid = make_embed_request(
+                    self.associated_item_identifier,
+                    property_traces
+                    + ".uuid",  # TODO: are we assuming the user will include ".uuid" or @id as part of prop trace?
+                    self.auth_key,
+                    single_item=True,
+                )
+                if not item_uuid:
+                    raise MetaWorkflowRunHandlerCreationError(
+                        f"Invalid property trace '{property_traces}' on item with the following ID: {self.associated_item_identifier}"
+                    )
+                return item_uuid
+
+
+            items_for_creation_uuids = []
+            for item_prop_trace in property_traces:
+                item_uuid = make_embed_request(
+                    self.associated_item_identifier,
+                    item_prop_trace
+                    + ".uuid",  # TODO: are we assuming the user will include ".uuid" or @id as part of prop trace?
+                    self.auth_key,
+                    single_item=True,
+                )
+                if not item_uuid:
+                    raise MetaWorkflowRunHandlerCreationError(
+                        f"Invalid property trace '{item_prop_trace}' on item with the following ID: {self.associated_item_identifier}"
+                    )
+                items_for_creation_uuids.append(item_uuid)
```

Comment:
I don't think this logic will work here, as you may receive multiple UUIDs, not just one, and you have to parse the result to obtain any UUIDs returned.